### PR TITLE
Fe/feature/component structure setup

### LINKED ISSUE #1: Create Wireframes
- [ ] Welcome Page
  - [ ] Profile Page (Show) 
  - [ ] Guild HomePage (Index)
  - [ ] Campaign Index Page

### LINKED ISSUE #4: Create PR Template

### LINKED ISSUE #5: Create a mockup of the website
Create folder and files for the required components
Install packages:

* react-router-dom
* cypress

### PR DESCRIPTION
**Pull request by okalimbet**
### Description
Add cypress test tool
Add react-route-dom for typescript

Currently users land on the homepage where they can see their profile information. On the top they can access 3 tabs that take users to Profile, users's quests log, or available quests. On the available quests page, users should be able to click on the quest to open a single quest page with the detailed description and available actions.

Add the following components:
* Common: Menu & MenuItem
* HomePage
* Quest
* QuestList
* User
* UserQuestLog

Add file with route paths
Add general ui components:
* Grid
* Button
* ActionButton

Add template for apiCalls

### Screenshots
Snip20210222_2
Snip20210222_3
Snip20210222_4
Snip20210222_5

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [ ] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [ ] I ran rubocop and fixed issues
- [ ] I ran full test suite - all tests passing 

### Other comments
This is a template structure for our website
Need to remove Accept buttons from the quests page - no longer a feature.
Tabs would be moved to the side of the page

### CODE REVIEW

File: src/modules/QuestsList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  match: object
+}
+
+
+type QuestProps = QuestList
+
+const QuestList: React.FC<QuestProps> = (props) => {
+  const {match} = props
+  const [currentQuest, setCurrentQuest] = useState<object>({})
+  // const id = [1, 2, 3]
```
Comment (CalebCyphers) [inline src/modules/QuestsList/index.tsx]: Can we remove this?

**Review by CalebCyphers — APPROVED** [review]

File: src/modules/QuestsList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  return (
+    <section style={{border: "1px solid black",
+    height: '90vh',
+    marginTop: 'auto',
+    marginBottom: 'auto',
+    padding: '1em',
+    display: 'flex',
+    flexDirection: "column",
+    justifyContent: "center"}}>
+      <h2>Availabel Quests</h2>
```
Comment (carson-jardine) [inline src/modules/QuestsList/index.tsx]: The spelling should be corrected to `Available`

**Review by carson-jardine — CHANGES_REQUESTED** [review]
This looks awesome (from what I can tell lol) just one minor spelling error I noticed that should be fixed before merging. Nice work!!!

**Review by c-bartell — APPROVED** [review]

**Review by smcunning — APPROVED** [review]
Great work on setting up the initial Front End! Really looking forward to seeing all you create!

**Review by jakeheft — APPROVED** [review]
There are a couple comments about a typo and comments left in but other than that everything looks great!

File: package.json [hunk 3]
```diff
@@ -22,7 +22,8 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "cypress": "cypress open"
```
Comment (GeorgieGirl24) [inline package.json:26]: Hooray, using cypress

File: src/modules/QuestsList/index.tsx (hunk 1 above)
Comment (okalimbet) [reply src/modules/QuestsList/index.tsx]: Done!

File: src/modules/QuestsList/index.tsx (hunk 2 above)
Comment (okalimbet) [reply src/modules/QuestsList/index.tsx]: Done!

**Review by GeorgieGirl24 — APPROVED** [review]

**Pull request merged**